### PR TITLE
Feature/net 7 minimal api

### PR DESCRIPTION
**Pull request by jeastham1993**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by Beau-Gosse-dev — CHANGES_REQUESTED** [review]
I like the disclaimer that asp.net isn't fully supported. We should make sure to communicate that.

File: src/NET7NativeMinimalAPI/ApiBootstrap/Function.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+var app = Startup.Build(args);
+
+var dataAccess = app.Services.GetRequiredService<ProductsDAO>();
+
+var cloudWatchClient = new AmazonCloudWatchLogsClient();
+
+app.MapGet("/", async (HttpContext context) =>
+{
+    app.Logger.LogInformation("Received request to list all products");
```
[the remark this answers is not in the document]
Comment (jeastham1993) [reply src/NET7NativeMinimalAPI/ApiBootstrap/Function.cs:22]: The other samples use the context.Logger object for logging, which we don't have access to here. I'm just using the built in ASP.NET logger.

File: src/NET7NativeMinimalAPI/ApiBootstrap/ApiBootstrap.csproj [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net7.0</TargetFramework>
+    <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+    <PublishReadyToRun>true</PublishReadyToRun>
+    <RootNamespace>GetProducts</RootNamespace>
+    <PublishAot>true</PublishAot>
+    <AssemblyName>bootstrap</AssemblyName>
+    <TrimMode>partial</TrimMode>
```
[the remark this answers is not in the document]
Comment (jeastham1993) [reply src/NET7NativeMinimalAPI/ApiBootstrap/ApiBootstrap.csproj:13]: Updated to use TrimMode full alongisde TrimmerRootAssembly.

File: src/NET7NativeMinimalAPI/ApiBootstrap/Function.cs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+    var products = await dataAccess.GetAllProducts();
+
+    app.Logger.LogInformation($"Found {products.Products.Count} products(s)");
+
+    context.Response.StatusCode = (int) HttpStatusCode.OK;
+    await context.Response.WriteAsJsonAsync(products);
```
[the remark this answers is not in the document]
Comment (jeastham1993) [reply src/NET7NativeMinimalAPI/ApiBootstrap/Function.cs:29]: No, I don't believe it does. I think TrimMode partial was cirumcventing it. I've updated to explicitly use the custom serializer.

**Pull request merged**